src/Bicep.Core/TypeSystem/Providers/ResourceTypesProviderDescriptor.cs

```diff
@@ -1,33 +1,47 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Bicep.Core.Registry;
+using Bicep.Core.Semantics.Namespaces;
+
 namespace Bicep.Core.TypeSystem.Providers;
 
-public record ResourceTypesProviderDescriptor
+public class ResourceTypesProviderDescriptor
```

Comment:
Out of interest, why switch from `record` -> `class`? Not questioning it, just curious.

Reply:
reverting to record